platforms/android/tangram/src/main/java/com/mapzen/tangram/TouchInput.java

```diff
@@ -330,6 +384,14 @@ private void setGestureDetected(Gestures g, boolean detected) {
     @Override
     public boolean onTouch(View v, MotionEvent event) {
 
+        if (event.getAction() == MotionEvent.ACTION_UP) {
```

Comment:
Should be `getActionMasked` https://developer.android.com/reference/android/view/MotionEvent.html#getActionMasked()